ouroboros-consensus/test/consensus-test/Test/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -594,6 +597,60 @@ runChainSync skew securityParam (ClientUpdates clientUpdates)
       in
       RelativeTime $ onset - unClockSkew skew
 
+    -- The target time (as reported by 'clientSystemTime') for the given
+    -- 'NewMaxSlot'.
+    clientTimeForNewMaxSlot :: NewMaxSlot -> RelativeTime
+    clientTimeForNewMaxSlot = \case
+        NewMaxClientSlot slot -> toOnset slot
+        NewMaxServerSlot slot -> toSkewedOnset slot
+
+        NewMaxClientAndServerSlot cslot sslot ->
+          toOnset cslot `max` toSkewedOnset sslot
+
+    clockUpdates :: Schedule NewMaxSlot
+    clockUpdates =
+        mkClockUpdates
+          (ClientUpdates clientUpdates)
+          (ServerUpdates serverUpdates)
+
+    clientTimeForTick :: Tick -> RelativeTime
```

Comment:
Maybe add a comment here that at least recommends rereading the module header for context on how ticks/time/skew/etc works in this test.